mrchecker-docs/documentation/Who-Is-MrChecker/Test-Framework-Modules/Core-Test-Module-Different-Environments.asciidoc

```diff
@@ -1,150 +1,133 @@
+
+**NOTE:** As environments.csv is a comma-separated file, please be careful while editing and then save it under Excel. 
+
+image::images/image082.png[]
+
+== Encrypting sensitive data
+
+Some types of data you might want to store as environment settings are sensitive in nature (e.g. passwords). You might not want to store them (at least not in their plaintext form) in your repository. To be able to encrypt sensitive data you need to do following:
+
+* 1 Create a secret (long, random chain of characters) and store it under *mrchecker-app-under-test\src\resources\secretData.txt*. Example: `LhwbTm9V3FUbBO5Tt5PiTUEQrXGgWrDLCMthnzLKNy1zA5FVTFiTdHRQAyPRIGXmsAjPUPlJSoSLeSBM`
+* 2 Exclude the file from being checked into the git repository by adding it to *git.ignore*. You will need to pass the file over a different channel among your teammates.
+* 3 Encrypt the values before putting them into the *environments.csv* file by creating following script (put the script where your jasypt library resides, e.g. *C:\MrChecker_Test_Framework\m2\repository\org\jasypt\jasypt\1.9.2*):
+
+----
+@ECHO OFF
+
+set SCRIPT_NAME=encrypt.bat
+set EXECUTABLE_CLASS=org.jasypt.intf.cli.JasyptPBEStringEncryptionCLI
+set EXEC_CLASSPATH=jasypt-1.9.2.jar
+if "%JASYPT_CLASSPATH%" == "" goto computeclasspath
+set EXEC_CLASSPATH=%EXEC_CLASSPATH%;%JASYPT_CLASSPATH%
+
+:computeclasspath
+IF "%OS%" == "Windows_NT" setlocal ENABLEDELAYEDEXPANSION
+FOR %%c in (%~dp0..\lib\*.jar) DO set EXEC_CLASSPATH=!EXEC_CLASSPATH!;%%c
+IF "%OS%" == "Windows_NT" setlocal DISABLEDELAYEDEXPANSION
+
+set JAVA_EXECUTABLE=java
+if "%JAVA_HOME%" == "" goto execute
+set JAVA_EXECUTABLE="%JAVA_HOME%\bin\java"
+
+:execute
+%JAVA_EXECUTABLE% -classpath %EXEC_CLASSPATH% %EXECUTABLE_CLASS% %SCRIPT_NAME% %*
+----
+* 4 Encrypt the values by calling
+
+----
+.\encrypt.bat input=someinput password=secret
+----
+
+image::images/image095.png[]
+
+* 5 Mark the value as encrypted by adding a prefix 'ENC(' and suffix ')' like: `ENC(JN3nOFol2GMZoUxR5z2wI2qdipcNH1UD)`
+
+image::images/image083.png[]
+
+== Bridge between external file nad Page class
+
+To map values from external file with Page class you ought to use class *GetEnvironmentParam*
+
+Therefore when you add new variable (row) in *environments.csv* you might need to add this variable to *GetEnvironmentParam*. 
+
+image::images/image084.png[]
+
+= Run test case with system environment
+
+To run test case with system environment, please use:
+* *-Denv=\<NameOfEnvironment\>*
+* *\<NameOfEnvironment\>* is taken as column name from file *mrchecker-app-under-test\src\test\resources\enviroments\environments.csv* 
+
+
+== Eclipse since mrchecker-core-module version 5.6.2.1
+Command Line
+
+----
+mvn test site -Dgroups=RegistryPageTestTag -Denv=DEV
+----
+
+
+image::images/image085.png[]
+image::images/image086_new.png[]
+
+
+== Eclipse prior to mrchecker-core-module version 5.6.2.1

```

Comment:
I think we decided to leave only one version of documentation and to divide it to "before version 5.6.2." and "after version 5.6.2."